rust/lance-table/src/format/manifest.rs

```diff
@@ -436,10 +436,16 @@ impl TryFrom<pb::Manifest> for Manifest {
 
         let data_storage_format = match p.data_format {
             None => {
-                if has_deprecated_v2_feature_flag(p.writer_feature_flags) {
-                    DataStorageFormat::new(LanceFileVersion::V2_0)
+                if fragments.is_empty() {
+                    // No fragments to inspect, best we can do is look at writer flags
+                    if has_deprecated_v2_feature_flag(p.writer_feature_flags) {
+                        DataStorageFormat::new(LanceFileVersion::Legacy)
+                    } else {
+                        DataStorageFormat::new(LanceFileVersion::Stable)
+                    }
```

Comment:
Isn't this backwards?

Reply:
Good point.  Let me make some migration tests for these cases

Reply:
Ok.  I added some regression test cases (confirmed they detected this) and switched the logic (now tests pass)